BlueprintUI/Sources/Element/UIViewElement.swift

```diff
@@ -0,0 +1,156 @@
+///     var text : String
+///     var font : UIFont
+///
+///     typealias UIViewType = UILabel
+///
+///     static func makeUIView() -> UILabel {
+///         UILabel()
+///     }
+///
+///     func updateUIView(_ view: UILabel) {
+///         view.text = self.text
+///         view.font = self.font
+///     }
+/// }
+/// ```
+public protocol UIViewElement : Element {
+    
+    /// The type of the view associated with the element.
+    associatedtype UIViewType : UIView
+        
+    /// Create and return a new instance of the provided view type.
+    ///
+    /// Note
+    /// ----
+    /// Ensure that you do not pass any values to the initializer of your view type
+    /// that you cannot also update in `updateUIView(_:)`, as view instances
+    /// are reused for sizing and measurement.
+    static func makeUIView() -> UIViewType
+
+    /// Update the view instance with the content from the element.
+    ///
+    /// Example
+    /// -------
+    /// If you were to implement a simple `UIViewElement` which wraps a `UILabel`,
+    /// your update method would look like this:
+    /// 
+    /// ```
+    /// func updateUIView(_ view: UILabel) {
+    ///     view.text = self.text
+    ///     view.font = self.font
+    /// }
+    /// ```
+    func updateUIView(_ view: UIViewType)
+    
+    /// Returns the sizing measurement for the element for the provided
+    /// measurement view.
+    ///
+    /// You usually do not need to implement this method – the default implementation of
+    /// this method simply calls `sizeThatFits(_:)` on the provided view.
+    ///
+    /// The view is fully configured and updated before this method is called – you do not need to
+    /// update it in any way.
+    ///
+    /// When To Override
+    /// ----------------
+    /// You may want to override this method if you need to mutate the value returned from `sizeThatFits(_:)`,
+    /// for example to round it up to an integral pixel value in the case of a `UILabel`, or if you
+    /// want to use some other sizing method like `systemLayoutSizeFitting(...)`.
+    func size(_ size : CGSize, thatFits view : UIViewType) -> CGSize
+}
+
+
+public extension UIViewElement {
+    
+    /// The default implementation simply forwards to `sizeThatFits(_:)`.
+    func size(_ size : CGSize, thatFits view : UIViewType) -> CGSize {
+        view.sizeThatFits(size)
+    }
+    
+    //
+    // MARK: Element
+    //
+    
+    /// Defer to the reused measurement view to provide the size of the element.
+    var content: ElementContent {
+        ElementContent {
+            UIViewElementMeasurer.shared.measure(element: self, in: $0)
+        }
+    }
+    
+    /// Provide the view for the element.
+    func backingViewDescription(bounds: CGRect, subtreeExtent: CGRect?) -> ViewDescription? {
+        UIViewType.describe {
+            $0.builder = {
+                Self.makeUIView()
+            }
+            
+            $0.apply {
+                self.updateUIView($0)
```

Comment:
I will continue to rally against the use of shorthand arguments in any block that's more than a single expression. This would be much clearer to anyone reading this code as:

```swift
UIViewType.describe { config in
  config.builder = Self.makeView
  config.apply { view in
    self.updateUIView(view)
  }
}
```

The shorthand isn't gaining anything here, it's just obfuscating the semantic meaning of arguments.

Reply:
I can change it, but again, it's pretty obvious what's happening from reading the code if you've ever seen Swift before...